src/pages/overlay/App.tsx

```diff
@@ -32,17 +32,34 @@ export default function App(){
     // Show/hide the overlay based on mouse movement
     const sleepTimer = useRef<NodeJS.Timeout | undefined>(undefined)
     const [sleeping, setSleeping] = useState(false)
+
+    // Allow children to know when we have been woken up
+    const [awoken, setAwoken] = useState<(() => void)[]>([])
+    const addAwoken = useCallback((callback: () => void) => {
+        setAwoken(current => [...current, callback])
+    }, [])
+    const removeAwoken = useCallback((callback: () => void) => {
+        setAwoken(current => current.filter(c => c !== callback))
+    }, [])
+    const awokenObj = useMemo(() => ({ add: addAwoken, remove: removeAwoken }), [addAwoken, removeAwoken])
+
+    // Wake the overlay for x milliseconds
     const wake = useCallback((time: number) => {
         setSleeping(false)
+        awoken.forEach(fn => fn())
         if (sleepTimer.current) clearTimeout(sleepTimer.current)
         sleepTimer.current = setTimeout(() => {
             setSleeping(true)
         }, time)
-    }, [])
+    }, [awoken])
+
+    // Immediately sleep the overlay
     const sleep = useCallback(() => {
         setSleeping(true)
         if (sleepTimer.current) clearTimeout(sleepTimer.current)
     }, [])
+
+    // When we unmount, clear the sleep timer
```

Comment:
can't this be done by adding a return to the use effect? if so, do that in the first use effect.

Reply:
Yah, I could add this to the first useEffect that handles loading the settings, but I think it's cleaner to keep them separate here -- you have the code that handles the settings, and the code that handles sleeping, not mixing concerns.